### PR TITLE
Dichroic filter implementation

### PR DESCRIPTION
**Pull request by JamesJieranShen**
Implements the dichroic filters, including placeholder data from the manufacturers. 
**Please Squash Merge!!** as the commit history may include unpublished data.

### CODE REVIEW

**Comment by JamesJieranShen** [comment]
I just added another commit to allow the optical filter files to be located outside of the rat code base. This is helpful for private data to be placed in experiment-specific code bases.

**Comment by JamesJieranShen** [comment]
@tannerbk @MorganAskins Hi Tanner and Morgan! Is this ready to be merged?

File: ratdb/OPTICS_Dichroic.ratdb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+{
+name: "OPTICS",
+index: "shortpass_dichroic_placeholder",
+valid_begin : [0, 0],
+valid_end : [0, 0],
+surface: 1,
+finish: "polished",
+model: "dichroic",
+type: "dielectric_dichroic",
+dichroic_property_file: "dichroic_data/knight_longpass_480nm_from_manufacturer.dat",
```
Comment (MorganAskins) [inline ratdb/OPTICS_Dichroic.ratdb:31]: shortpass data file

**Comment by MorganAskins** [comment]
Testing the filters in the validation geometry with Geant4 11.2.0

### High pass filter, long wavelength photon bomb
image

### High pass filter, short wavelength photon bomb
image

Implementation looks to be working and the logs are not spammed so the redirect is fine. Will merge soon.

File: ratdb/OPTICS_Dichroic.ratdb (hunk 1 above)
Comment (JamesJieranShen) [reply ratdb/OPTICS_Dichroic.ratdb]: Fixed!

**Pull request merged**